Aikido.Zen.Core/ReportingStatus.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using Aikido.Zen.Core.Models.Events;
+
+namespace Aikido.Zen.Core
+{
+    internal class ReportingStatus
+    {
+        private DateTime? _lastReported;
+        private bool _success;
+        private readonly TimeSpan _gracePeriod = TimeSpan.FromSeconds(30);
+
+        public void SignalReporting(string operation, bool success)
+        {
+            // The last reported time and success status can be set in 2 cases:
+            // 1. Initial reporting (Started event)
+            // 2. Heartbeat event
+
+            if (operation != Heartbeat.HeartbeatEventName && operation != Started.StartedEventName)
```

Comment:
```suggestion
            if (operation != Heartbeat.EventType && operation != Started.EventType)
```

Reads a bit nicer imo? or `EventName` etc?